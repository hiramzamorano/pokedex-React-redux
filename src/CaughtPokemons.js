import React from 'react';
import {
    Header,
    Item
} from 'semantic-ui-react';

export default class CaughtPokemons extends React.Component {
    render() {
        return (
            <React.Fragment>
                <Header as="h2">Caught Pokemons</Header>
                <Item.Group>
                    <Item>
                        <Item.Image size='small' src={`https://api.adorable.io/avatars/80/pikachu.png`} />
                        <Item.Content>
                            <Item.Header>Pikachu</Item.Header>
                            <Item.Description>Electric type</Item.Description>
                        </Item.Content>
                    </Item>
                </Item.Group>
            </React.Fragment>
        );
    }
}