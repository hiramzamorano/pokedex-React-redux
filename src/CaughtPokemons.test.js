import React from 'react';
import { render } from '@testing-library/react';
import { CaughtPokemons } from './CaughtPokemons';

describe('CaughtPokemons', () => {
    it('should render a list of pokemons', () => {
        const pokemons = [
            {
                id: 132,
                name: 'ditto',
                types: [{type: {name: 'normal'}}] ,
                height: 3,
                weight: 40,
                isCaught: true,
                sprites: {
                    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png'
                }
            },
            {
                id: 92,
                name: 'gastly',
                types: [{type: {name: 'normal'}}],
                height: 13,
                weight: 1,
                isCaught: true,
                sprites: {
                    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/92.png'
                }
            }
        ];

        const component = render(<CaughtPokemons pokemons={pokemons} />);
        expect(component).toMatchSnapshot();
    })

    it('should display no captured pokemons when pokemon list is empty', () => {
        const pokemons = [];
        const component = render(<CaughtPokemons pokemons={pokemons} />);
        expect(component.baseElement).toHaveTextContent('No pokemons captured yet');
        expect(component).toMatchSnapshot();
    })
});