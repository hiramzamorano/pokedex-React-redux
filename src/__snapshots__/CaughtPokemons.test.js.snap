// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaughtPokemons should display no captured pokemons when pokemon list is empty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="ui header"
      >
        Caught Pokemons
      </h2>
      <h3
        class="ui header"
        style="text-align: center;"
      >
        No pokemons captured yet
      </h3>
      <div
        class="ui items"
      />
    </div>
  </body>,
  "container": <div>
    <h2
      class="ui header"
    >
      Caught Pokemons
    </h2>
    <h3
      class="ui header"
      style="text-align: center;"
    >
      No pokemons captured yet
    </h3>
    <div
      class="ui items"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CaughtPokemons should render a list of pokemons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="ui header"
      >
        Caught Pokemons
      </h2>
      <div
        class="ui items"
      >
        <div
          class="item"
        >
          <div
            class="ui tiny image"
          >
            <img
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png"
            />
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              #
              132
               
              ditto
            </div>
            <div
              class="description"
            >
              Type: 
              normal
            </div>
            <div
              class="description"
            >
              Height: 
              3
              , Weight: 
              40
            </div>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="ui tiny image"
          >
            <img
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/92.png"
            />
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              #
              92
               
              gastly
            </div>
            <div
              class="description"
            >
              Type: 
              normal
            </div>
            <div
              class="description"
            >
              Height: 
              13
              , Weight: 
              1
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h2
      class="ui header"
    >
      Caught Pokemons
    </h2>
    <div
      class="ui items"
    >
      <div
        class="item"
      >
        <div
          class="ui tiny image"
        >
          <img
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png"
          />
        </div>
        <div
          class="content"
        >
          <div
            class="header"
          >
            #
            132
             
            ditto
          </div>
          <div
            class="description"
          >
            Type: 
            normal
          </div>
          <div
            class="description"
          >
            Height: 
            3
            , Weight: 
            40
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="ui tiny image"
        >
          <img
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/92.png"
          />
        </div>
        <div
          class="content"
        >
          <div
            class="header"
          >
            #
            92
             
            gastly
          </div>
          <div
            class="description"
          >
            Type: 
            normal
          </div>
          <div
            class="description"
          >
            Height: 
            13
            , Weight: 
            1
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
